src/forms/renderers/OAuth.tsx

```diff
@@ -54,11 +53,22 @@ const OAuthproviderRenderer = ({
         [data, dataKeys, descriminatorProperty, requiredFields]
     );
 
-    const provider = options ? options[Options.oauthProvider] : NO_PROVIDER;
-    const capitalizedProvider = useMemo(() => startCase(provider), [provider]);
+    const providerVal = options ? options[Options.oauthProvider] : NO_PROVIDER;
+    const provider = useMemo(() => startCase(providerVal), [providerVal]);
 
-    const onError = (error_: any) => {
-        setErrorMessage(error_);
+    const onError = (error: any) => {
+        if (error === 'access_denied') {
+            setErrorMessage(
+                intl.formatMessage(
+                    {
+                        id: 'oauth.authentication.denied',
+                    },
+                    { provider }
+                )
+            );
```

Comment:
if a user denies access for some reason we're catching it so that we can message to them. Other errors seem to be pretty solid.